### DISCUSSION TITLE
Add-In effects support ? Quadrilateral Correction

### DISCUSSION THREAD

**Discussion by Mac-EGM** [PintaProject/Pinta#890] [Ideas]
I came to Pinta for My Linux Devices from Paint.net my favourite Windows Image tool. In Paint.net the embedded scripting language engine allows many to make their own add-ins. My absolute go to is Quadrilateral Correction so most images taken on a phone can be stretched to be FLAT and used as say Signs or backgrounds in other works. I am working through family pictures submitted and in poor shape starting with this.  I do not know as yet enough about Pinta to understand how difficult it would be to modify this Paint.net app to work with Pinta but I can say it would be a welcomed addition for myself and I suggest many others.  See https://forums.getpaint.net/topic/110247-quadrilateral-correction-v131-dec-16-2021/  & GitHub as there appears to be a correlation between the packages and similar needs and questions.

**Comment by cameronwhite** [comment]
There will be support for add-ins in the next release of Pinta, but some work would be needed to modify a Paint.NET addin for Pinta. (Both are written in C#, but the API for Pinta's effects is not the same)

**Comment by Mac-EGM** [comment]
Thanks for the update. I see the API as the least of the problem if you
know your way around the code base while access to the core algorythm to
stretch the image is the hardest part! Is the code accessible on GIT hub ?
Add in or native its a brilliant tool which is well worth investigation.

regards

Mark Eades
